internal/pkg/service/appsproxy/proxy/pagewriter/error.go

```diff
@@ -0,0 +1,111 @@
+package pagewriter
+
+import (
+	"net"
+	"net/http"
+	"strings"
+
+	svcerrors "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type errorPageData struct {
+	Status      int
+	StatusText  string
+	Messages    []string
+	Details     string
+	ExceptionID string
+}
+
+func (pw *Writer) ProxyErrorHandler(w http.ResponseWriter, req *http.Request, err error) {
+	var dnsError *net.DNSError
+	if errors.As(err, &dnsError) {
+		pw.logger.Info(req.Context(), "app is not running, rendering spinner page")
+		pw.WriteSpinnerPage(w, req)
+	} else {
+		err = svcerrors.NewBadGatewayError(err).WithUserMessage("Request to application failed.")
+		pw.WriteError(w, req, err)
+	}
+}
+
+func (pw *Writer) WriteError(w http.ResponseWriter, req *http.Request, err error) {
+	// Status code, by default 500
+	status := http.StatusInternalServerError
+	var statusProvider svcerrors.WithStatusCode
+	if errors.As(err, &statusProvider) {
+		status = statusProvider.StatusCode()
+	}
+
+	// Error name, by default "internal"
+	var errName string
+	var errNameProvider svcerrors.WithName
+	if errors.As(err, &errNameProvider) {
+		errName = errNameProvider.ErrorName()
+	} else {
+		errName = "internal"
+	}
```

Comment:
```suggestion
	var errName := "internal"
	var errNameProvider svcerrors.WithName
	if errors.As(err, &errNameProvider) {
		errName = errNameProvider.ErrorName()
	}
```

Reply:
It doesn;t match style of the other blocks, default value may require some logic/code.